merge ansible role octavia_preconf as per jira OSPC-189

puni4220 opened the pull request:
this PR is being raised for jira [1] to merge the ansible role to automate the required pre-requisites for enabling the amphora provider for octavia. The ansible role currently has tasks for:
1. creating the lb-mgmt-net and lb-mgmt-subnet
2. creating the required security groups
3. creating the ssh keypair for logging into amphora 
4. creating the required certificates for the amphora provider
5. creating the required helm amphora values file to enable the amphora provider
6. creating the required flavor for the amphora instances
7. modifying the quotas for the admin tenant to accomodate amphorae
8. creating the health-manager ports

the role can run from the bastion node and currently has a required to use a shell script because the openstack.cloud.port module doesn't include "--host" argument which is required for creating health-manager ports. It should also be note that currently the ansible role creates health-manager ports for nodes labelled as "openstack-control-plane=enabled" 

I will be raising a separate PR for the playbook to run this ansible role and a document

[1] https://rackspace.atlassian.net/browse/OSPC-189

japerezjr reviewed (approved):
Tested in lab and worked as expected.

puni4220 commented:
Hello, I've removed the tests directory from the ansible role to prevent the CI jobs from trying to run the role and modified the shell scripts accordingly

cloudnull commented on ansible/roles/octavia_preconf/README.md:
````diff
@@ -0,0 +1,90 @@
+```
+(octavia_preconf) root@saturn-c1:~# cat octavia-preconf-main.yaml 
+
+- name: Pre-requisites for enabling amphora provider in octavia
````
roles can have playbooks within them, under the playbooks directory. I'd recommend adding a playbook there, if not in the common playbooks directory found in https://github.com/rackerlabs/genestack/tree/main/ansible/playbooks

puni4220 replied:
Sure; I am planning to add a playbook for the role in a separate PR altogether

cloudnull commented on ansible/roles/octavia_preconf/README.md:
```diff
@@ -0,0 +1,90 @@
+- name: Pre-requisites for enabling amphora provider in octavia
+  hosts: localhost
+  environment:
+    OS_ENDPOINT_TYPE: publicURL
```
it'd be great if we could use the k8s lookup in the playbook to automatically set many of these options.

puni4220 replied:
Will try to include the lookup plugins wherever possible to set the variables; I will update the README for the role accordingly when I raise a PR for the role playbook

cloudnull commented on ansible/roles/octavia_preconf/templates/octavia_amphora_provider.yaml.j2:
is this something we should add to our kustomize as the default?

puni4220 replied:
hey Kevin; I am not sure how that would work if we were to add it to kustomize; we do set values in this file on the fly by obtaining the UUID(s) of the resources created by the role (flavor, network) and also the password for the keys; the role does generate a debug message at the end which talks about including this file along with the other helm overrides to enable amphora provider

cloudnull commented on ansible/roles/octavia_preconf/README.md:
```diff
@@ -0,0 +1,90 @@
+This is a role for performing the pre-requisite tasks to enable amphora provider for octavia for example creating a network and a subnet for amphorae, uploading the image to glance, creating ssh keys etc. This is mainly intended for running octavia in a k8s environment
+
+Requirements
+------------
```
I think this is rst in an md file?
```suggestion
## Requirements 
```

cloudnull commented on ansible/roles/octavia_preconf/README.md:
```diff
@@ -0,0 +1,90 @@
+3. It is recommended to create a virtual environment to run this role; steps will be shared below
+
+Creating a virtual environment
+------------
```
same as above.

cloudnull commented on ansible/roles/octavia_preconf/README.md:
```diff
@@ -0,0 +1,90 @@
+Refer to the role defaults for more detail  
+
+Dependencies
+------------
```
same as above.

cloudnull commented on ansible/roles/octavia_preconf/README.md:
```diff
@@ -0,0 +1,90 @@
+The role has no external dependencies; only the steps shared above for creating the virutal environment are required
+
+Example Playbook
+----------------
```
same as above.

cloudnull reviewed (changes requested):
This is great. Just a couple nits on the readme and a few changes to make the linter happy. I would also like to see the practical application of this role documented within our octavia operations guide.

puni4220 commented on ansible/roles/octavia_preconf/README.md:
```diff
@@ -0,0 +1,90 @@
+octavia_preconf
+=========
```
done; fixed in the push

puni4220 commented on ansible/roles/octavia_preconf/README.md:
```diff
@@ -0,0 +1,90 @@
+note that the kubeconfig in this step has been copied from the master node and it should be modified accordingly
+
+Role Variables
+--------------
```
commited and pushed

rackerchris reviewed (approved):
LGTM +1
